services/QuillLMS/app/controllers/stripe_integration/subscription_renewals_controller.rb

```diff
@@ -2,10 +2,14 @@
 
 module StripeIntegration
   class SubscriptionRenewalsController < ApplicationController
+    class UpdatingCanceledSubscriptionError < StandardError; end
+
     def create
+      raise UpdatingCanceledSubscriptionError if stripe_subscription_canceled?
+
       Stripe::Subscription.update(stripe_subscription_id, cancel_at_period_end: cancel_at_period_end)
       render json: {}, status: 200
-    rescue Stripe::InvalidRequestError => e
+    rescue UpdatingCanceledSubscriptionError, Stripe::InvalidRequestError => e
```

Comment:
Is this just to track these? Since the outcome (a returned `500` error)

Reply:
This should be an error, but I would be surprised if this ever fires since it's a very weird edge case (user tries to update renewal settings on a subscription after canceling the subscription on stripe UI).   It happened previously since it was dependent on another webhook that wasn't functioning properly.